commit f0df9513124603af6b8ee7d11eca5fbf3b3c01d9 (refs/changes/34/334/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-08-22T18:27:34-07:00 (3 years, 2 months ago)
    
    AGENT-1029 CN agent reports negative disk_system_used_bytes values in sysinfo
    Reviewed by: Joshua M. Clulow <jmc@joyent.com>

diff --git a/lib/heartbeater.js b/lib/heartbeater.js
index 24f303a..c8b7a92 100644
--- a/lib/heartbeater.js
+++ b/lib/heartbeater.js
@@ -518,19 +518,11 @@ StatusReporter.prototype.gatherDiskUsage = function (vms, callback) {
         },
         function (cb) {
             // #7
-            zpool.list(
-                'zones', { parseable: true, fields: ['size', 'alloc'] },
-                function (error, fields, pools) {
-                    if (error) {
-                        cb(error);
-                        return;
-                    }
-                    usage.pool_size_bytes = toInt(pools[0][0]);
-                    usage.pool_alloc_bytes = toInt(pools[0][1]);
-                    cb();
-                });
-        },
-        function (cb) {
+            var poolds = datasets['zones'];
+            usage.pool_alloc_bytes = toInt(poolds.used);
+            usage.pool_size_bytes = toInt(poolds.used) +
+                toInt(poolds.available);
+
             // #8
             // All separated usages should be subtracted from the allocated
             // here. Anything not specifically listed is treated as 'system
